backend/app/views.py

```
@@ -1277,56 +1373,139 @@ def metadata_view(request, instance):
+                portal=portal,
+                require_credentials=True
+            )
+
+            if form.is_valid():
+                username = form.cleaned_data['username']
+                password = form.cleaned_data['password']
+
+                # Test connection
+                logger.debug(f"Testing auth for {portal.alias} with user {username[0] if username else ''}...")
+                auth_payload = {'username': username, 'password': password, 'url': portal.url}
+                auth_result = utils.try_connection(auth_payload)
```

Ruff (0.13.1)

`utils` may be undefined, or defined from star imports

(F405)